jsapp/js/account/organization/InviteModal.tsx

```diff
@@ -1,22 +1,105 @@
-import { Group, Modal, Stack, Text, TextInput, ModalProps } from '@mantine/core'
+import type { ModalProps } from '@mantine/core'
+import { Group, Modal, Stack, Text, TextInput, Loader } from '@mantine/core'
 import ButtonNew from 'jsapp/js/components/common/ButtonNew'
 import { Select } from 'jsapp/js/components/common/Select'
+import { useSendMemberInvite } from './membersInviteQuery'
+import { useState } from 'react'
+import { OrganizationUserRole } from './organizationQuery'
+import userExistence from 'js/users/userExistence.store'
+import { useField } from '@mantine/form'
+import { checkEmailPattern, notify } from 'js/utils'
 
 export default function InviteModal(props: ModalProps) {
+  const inviteQuery = useSendMemberInvite()
+
+  const [role, setRole] = useState<string | null>(null)
+
+  async function handleUsernameOrEmailCheck(value: string) {
+    if (value === '' || checkEmailPattern(value)) {
+      return null
+    }
+
+    const checkResult = await userExistence.checkUsername(value)
+    if (checkResult === false) {
+      return t('This username does not exist. Please try again.')
+    } else {
+      return null
+    }
+  }
+
+  const userOrEmail = useField({
+    initialValue: '',
+    validate: handleUsernameOrEmailCheck,
+    validateOnBlur: true,
+  })
+
+  const handleSendInvite = () => {
+    if (role) {
+      inviteQuery
+        .mutateAsync({
+          invitees: [userOrEmail.getValue()],
+          role: role as OrganizationUserRole,
+        })
+        .then(() => {
+          userOrEmail.reset()
+          setRole(null)
+          props.onClose()
+        })
+        .catch((error) => {
+          if (error.responseText && JSON.parse(error.responseText)?.invitees) {
+            notify(JSON.parse(error.responseText)?.invitees.join(), 'error')
+          } else {
+            notify(t('Failed to send invite'), 'error')
+          }
+        })
+    }
+  }
+
+  const handleClose = () => {
+    userOrEmail.reset()
+    setRole(null)
+    props.onClose()
+  }
+
+  const isValidated = role !== null && userOrEmail.isDirty() && !userOrEmail.isValidating && !userOrEmail.error
```

Comment:
```suggestion
  const isValidated = !!role && userOrEmail.isDirty() && !userOrEmail.isValidating && !userOrEmail.error
```
Just making this a little easier to read